### PR TITLE
Fix: C code generator should provide `const uint8_t *` as 4th argument to  `mol_union_builder_initialize`

### PR DESCRIPTION
**Pull request by eval-exec**
This PR fix a bug in C code generator, and added some test case.

### CODE REVIEW

**Comment by eval-exec** [comment]
I just added a test case for Union type:
```
union UnionC {
    Word,
    byte,
}
```
and the [CI action](https://github.com/nervosnetwork/molecule/actions/runs/4280445085/jobs/7452296886) failed on `UnionC`, because `MOLECULE_API_DECORATOR void mol_union_builder_initialize(mol_builder_t *builder, mol_num_t data_capacity, mol_num_t item_id, const uint8_t *default_ptr, mol_num_t default_len)` want `const uint8_t*` on `4th` argument, but the generated code provide `const uint8_t (*)[2]`

**Comment by code-monad** [comment]
> ```c
> (const uint8_t *)&MolDefault_Word
> ```

LGTM as a C code.

File: tools/codegen/src/generator/languages/c/builder.rs [hunk 1]
```diff
@@ -110,8 +110,7 @@ impl GenBuilder for ast::Union {
             let name = if default.is_byte() {
                 "NULL".to_owned()
             } else {
-                let name = default.default_constant();
-                format!("&{}", name)
+                default.default_constant()
```
Comment (eval-exec) [inline tools/codegen/src/generator/languages/c/builder.rs:113]: The C code of `default.default_constant()` is something like 
```C
MOLECULE_API_DECORATOR const uint8_t MolDefault_Word[2] = {____, ____};
```
the `mol_union_builder_initialize`'s 4th argument accept `const uint8_t *`, so, it's OK to pass `MolDefault_World` (type: `const uint8_t *`) to `mol_union_builder_initialize`;

---------------
In history commit, we pass `&MolDefault_UnionA` (type: `const uint8_t (*)[2]`) to `mol_union_builder_initialize`, so, it won't compile successfully.

Comment (eval-exec) [reply tools/codegen/src/generator/languages/c/builder.rs:113]: This PR is ready for review.
@yangby-cryptape @driftluo @code-monad

**Review by code-monad — APPROVED** [review]

**Review by driftluo — APPROVED** [review]

**Review by yangby-cryptape — APPROVED** [review]

**Pull request merged**